### ISSUE TITLE
Converting ADL schemes into Sql and generating tables

### ISSUE THREAD

**Issue by fedkh** [openEHR/adl-tools#10]
Hello,

I am having difficulties in converting an archetype into Sql scheme in order to generate the specific tables.
Can anyone please help me understand this further or give me some guidelines.

Thanks a lot

My regards,
Fadoua

**Comment by ppazos** [comment]
In general is better to have a data model based on the information model not on the archetypes, since structures of the IM are shared between different archetypes. Generating an schema from archetypes is a bad idea because it fragments the IM and makes querying more difficult.

**Comment by wolandscat** [comment]
@fedkh please see the comment from @ppazos . I don't think any openEHR developer does what you are suggesting, for the reasons he states.

**Comment by fedkh** [comment]
Hello @ppazos @wolandscat  thank you for the response, although it has been a year when i posted the issue :D...
Anyway, i came across a research paper, which introduces a mapping between relational database and archetypes that's why i wanted to understand clearly the idea here is the paper: "Archetype relational mapping - a practical openEHR persistence solution".
Thanks

**Comment by ppazos** [comment]
@fedkh there is some work done in China using that approach, I talked with members of that team and explained why it was not the best approach in general, but it might be useful for small / test systems. Not recommended for big / complex / production systems, querying can get chaotic, since you'll need to get the same data e.g. blood pressure, from different tables. And will difficult database management, since adding new archetypes to the system, will add new tables to the DB.

**Comment by fedkh** [comment]
@ppazos yes agree with you thank you a lot for the clarification. Moving toward Nosql schema will be more convenient in this context.

**Comment by ppazos** [comment]
@fedkh not recommending to generate SQL schema from archetypes is not the same as recommending a document or OO database. You can use SQL generating the schema/data model from the openEHR IM. My CDR does that: https://github.com/ppazos/cabolabs-ehrserver
